### PR TITLE
 Reduce NullAway suppressions in throwing code

### PR DESCRIPTION
**Pull request by JarvisCraft**
Instead of using `throwFoo(); return null;` with suppressed `NullAway` lint just do `throw newFoo()`.

This removes the need for `NullAway` suppressions since now Java type system is happy deducing the return type of the methods to the bottom type which coerces to anything.

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit-framework/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Method [preconditions](https://docs.junit.org/snapshot/api/org.junit.platform.commons/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [x] [Coding conventions](https://github.com/junit-team/junit-framework/blob/HEAD/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit-framework/blob/HEAD/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [x] Public API has [Javadoc](https://github.com/junit-team/junit-framework/blob/HEAD/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [x] Change is documented in the [User Guide](https://docs.junit.org/snapshot/) and [Release Notes](https://docs.junit.org/snapshot/release-notes.html) (the change is effectively internal)

### CODE REVIEW

File: junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertInstanceOf.java [hunk 1]
```diff
@@ -46,18 +46,17 @@ static <T> T assertInstanceOf(Class<T> expectedType, @Nullable Object actualValu
...
-			assertionFailure() //
+			throw assertionFailure() //
 					.message(messageOrSupplier) //
 					.reason(actualValue == null ? "Unexpected null value" : "Unexpected type") //
 					.expected(expectedType) //
 					.actual(actualValue == null ? null : actualValue.getClass()) //
 					.cause(actualValue instanceof Throwable t ? t : null) //
 					.trimStacktrace(Assertions.class) //
-					.buildAndThrow();
+					.build();
```
Comment (marcphilipp) [inline junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertInstanceOf.java]: I've just pushed 85d360fc87cbf82f4515ad03579c1765bcbfc463 which makes the change to this file unnecessary.

File: junit-jupiter-api/src/main/java/org/junit/jupiter/api/Assertions.java [hunk 2]
```diff
@@ -117,10 +117,9 @@ protected Assertions() {
 	 * generic return type {@code V}.
 	 */
 	@Contract(" -> fail")
-	@SuppressWarnings({ "NullAway", "TypeParameterUnusedInFormals" })
+	@SuppressWarnings({ "TypeParameterUnusedInFormals" })
```
Comment (marcphilipp) [inline junit-jupiter-api/src/main/java/org/junit/jupiter/api/Assertions.java]:
```suggestion
	@SuppressWarnings("TypeParameterUnusedInFormals")
```

and so on below

**Review by marcphilipp — CHANGES_REQUESTED** [review]
Thanks for the PR! 👍

File: junit-jupiter-api/src/main/java/org/junit/jupiter/api/Assertions.java (hunk 2 above)
Comment (JarvisCraft) [reply junit-jupiter-api/src/main/java/org/junit/jupiter/api/Assertions.java]: Good catch; fixed it

File: junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertInstanceOf.java (hunk 1 above)
Comment (JarvisCraft) [reply junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertInstanceOf.java]: I've rebased on master but I would still suggest keeping the explicit `throw` here since it operates on Java's type system level (e.g. it participates in reachability checks of javac, a la guaranteed local var initialization) which is less prone to breakages and more obvious when reviewing the code.

**Review by marcphilipp — APPROVED** [review]
